gui/wxpython/lmgr/frame.py

```diff
@@ -1172,6 +1166,27 @@ def ChangeMapset(self, mapset):
                 display.SetTitleWithName(str(dispId))  # TODO: signal ?
                 dispId += 1
 
+    def SwitchMapset(self, grassdb, location, mapset):
+        """Switch to location and mapset"""
+        gisenv = grass.gisenv()
+        if can_switch_mapset_interactive(self, grassdb, location, mapset):
+            # Switch to mapset in the same location
+            if (grassdb == gisenv['GISDBASE'] and location == gisenv['LOCATION_NAME']):
+                self.datacatalog.changeMapset.emit(mapset=mapset)
+            # Switch to mapset in the same grassdb
+            elif grassdb == gisenv['GISDBASE']:
+                self.datacatalog.changeLocation.emit(mapset=mapset,
+                                                     location=location,
+                                                     dbase=None)
+            # Switch to mapset in a different grassdb
+            else:
+                self.datacatalog.changeLocation.emit(mapset=mapset,
+                                                     location=location,
+                                                     dbase=grassdb)
+            self.datacatalog.tree.UpdateCurrentDbLocationMapsetNode()
+            self.datacatalog.tree.ExpandCurrentMapset()
+            self.datacatalog.tree.RefreshItems()
```

Comment:
Layer manager shouldn't be calling methods of tree directly like that, this would at least need to be wrapped in some function in tree.py or catalog.py, and it's code duplication with code in tree.py.